dsl/src/main/kotlin/com/chutneytesting/kotlin/junit/engine/ChutneyExecutionContexts.kt

```diff
@@ -0,0 +1,281 @@
+    CONFIG_REPORT_ROOT_PATH("junit.chutney.report.rootPath"),
+    CONFIG_CONSOLE_LOG_COLOR("junit.chutney.log.console.color")
+}
+
+enum class ChutneyJUnitReportingKeys(val value: String) {
+    REPORT_JSON_STRING("chutney.report"),
+    REPORT_STEP_JSON_STRING("chutney.report.step")
+}
+
+class ChutneyEngineExecutionContext(private val request: ExecutionRequest) {
+    private val executionService = ExecutionService()
+    private val scenarioExecutions = HashMap<Long, ChutneyScenarioExecutionContext>()
+    private val startLatches = HashMap<TestDescriptor, CountDownLatch>()
+    private val syncExecutionSemaphore: Semaphore = Semaphore(1)
+    private val endExecutionLatch: CountDownLatch = CountDownLatch(request.rootTestDescriptor.children.size)
+
+    private var startScenarioDisposable: Disposable
+    private var beginStepDisposable: Disposable
+    private var endStepDisposable: Disposable
+    private var endScenarioDisposable: Disposable
+
+    init {
+        val chutneyBus = RxBus.getInstance()
+        startScenarioDisposable =
+            chutneyBus.register(StartScenarioExecutionEvent::class.java, this::startScenarioExecution)
+        beginStepDisposable =
+            chutneyBus.register(BeginStepExecutionEvent::class.java, this::beginStepExecution)
+        endStepDisposable = chutneyBus.register(EndStepExecutionEvent::class.java, this::endStepExecution)
+        endScenarioDisposable = chutneyBus.register(EndScenarioExecutionEvent::class.java, this::endScenarioExecution)
+    }
+
+    fun startExecution() {
+        request.engineExecutionListener.executionStarted(request.rootTestDescriptor)
+    }
+
+    fun executeScenario(scenarioDescriptor: ChutneyScenarioDescriptor) {
+        syncExecutionSemaphore.acquire()
+
+        val latch = CountDownLatch(1)
+        startLatches[scenarioDescriptor] = latch
+
+        try {
+            val executionId = executionService.execute(
+                scenarioDescriptor.chutneyScenario,
+                resolveEnvironmentName(scenarioDescriptor)
+            )
+            scenarioExecutions[executionId] = ChutneyScenarioExecutionContext(request, scenarioDescriptor)
+        } catch (t: Throwable) {
+            request.engineExecutionListener.executionFinished(
+                scenarioDescriptor,
+                testExecutionResultFromStatus(
+                    throwable = convertExecuteException(t, scenarioDescriptor),
+                    status = arrayOf(Status.FAILURE)
+                )
+            )
+            endExecutionLatch.countDown()
+        } finally {
+            latch.countDown()
+        }
+    }
+
+    fun endExecution() {
+        endExecutionLatch.await()
+        unregisterRxBus()
+
+        request.engineExecutionListener.executionFinished(
+            request.rootTestDescriptor,
+            TestExecutionResult.successful() // global status could be executionStatus() but brings some integration problems with ide
```

Comment:
I don't understand those lines. Is it really used for something ? I don't see any different behavior when removing them